_posts/2019-01-30-get-started-libaktualizr.adoc

```diff
@@ -0,0 +1,264 @@
+in the command queue until Resume() is issued.
+Commands that are issued after Pause() will be put on a command queue,
+but not executed until Resume() is called.
+
+The Pause() function returns immediately, while pausing the running command
+still may be in progress.
+
+The function has no effect in case execution was already paused.
+
+[source,cpp]
+void Aktualizr::Resume()
+
+Resumes the execution of a previously paused command and all subsequent commands
+in the command queue.
+Returns immediately. The function has no effect if the execution was not paused.
+
+[source,cpp]
+void Aktualizr::Abort()
+
+Requests the currently running command to abort and flushes the command queue.
+The Abort() function will block until the command queue is empty and no command
+is in executed state.
+
+Can also be called on a previously paused instance, but doesn't change
+it's paused state.
+If a paused command was aborted, next time it will start the execution
+from the place it was paused.
+
+[source,cpp]
+Aktualizr::~Aktualizr(Config &config)
+
+Calls Abort() and destroys the aktualizr object.
+
+==== Campaign management commands
+[source,cpp]
+std::future<result::CampaignCheck> Aktualizr::CampaignCheck()
+
+Check for campaigns.
+Campaigns are a concept outside of Uptane, and allow for user approval of
+updates before the contents of the update are known.
+
+[source,cpp]
+std::future<void> Aktualizr::CampaignAccept(const std::string &campaign_id)
+
+Accept a campaign for the current device.
+Campaigns are a concept outside of Uptane, and allow for user approval of
+updates before the contents of the update are known.
+
+==== Update management commands
+[source,cpp]
+std::future<void> Aktualizr::SendDeviceData()
+
+Sends local device data to the server.
+This includes network status, installed packages, hardware etc.
+
+[source,cpp]
+std::future<result::UpdateCheck> Aktualizr::CheckUpdates()
+
+Fetch Uptane metadata and check for updates.
+This collects a client manifest, PUTs it to the director, updates the
+Uptane metadata (including root and targets), and then checks the metadata
+for target updates.
+
+[source,cpp]
+std::future<result::Download> Aktualizr::Download(const std::vector<Uptane::Target> &updates)
+
+Download targets specified in the input vector that were previously
+returned by CheckUpdates.
+
+[source,cpp]
+std::future<result::Install> Aktualizr::Install(const std::vector<Uptane::Target> &updates)
+
+Install previously downloaded targets.
+
+[source,cpp]
+std::ifstream Aktualizr::GetStoredTarget(const Uptane::Target &target)
+
+==== Miscellaneous commands
+[source,cpp]
+void Aktualizr::UptaneCycle()
+
+Synchronously run an uptane cycle: check for updates, download any new
+targets, install them, and send a manifest back to the server.
+
+[source,cpp]
+std::future<void> Aktualizr::RunForever()
+
+Asynchronously run aktualizr indefinitely until Shutdown is called.
```

Comment:
I noticed this is the line from the doxygen but it's maybe too vague. Something like: "Automatic check and install updates indefinitely: runs UptaneCycle() in a loop at regular intervals until Shutdown() is called" ?

Reply:
Yes, that sounds better. Except that here we don't have the Shutdown call :)